pallets/phala/src/mining.rs

```diff
@@ -568,10 +566,20 @@ pub mod pallet {
 		///
 		/// We really don't want to crash the interrupt the message processing. So when there's an
 		/// error we return it, and let the caller to handle it gracefully.
-		fn try_handle_settle(info: &SettleInfo, now: u64) -> DispatchResult {
+		fn try_handle_settle(info: &SettleInfo, now: u64, emit_ts: u64) -> DispatchResult {
 			if let Some(account) = WorkerBindings::<T>::get(&info.pubkey) {
 				let mut miner_info = Self::miners(&account).ok_or(Error::<T>::MinerNotFound)?;
 				debug_assert!(miner_info.state.can_settle(), "Miner cannot settle now");
+				if miner_info.v_updated_at >= emit_ts {
```

Comment:
@kvinwang Please carefully check if this works

Reply:
Look's good